lib/rules/no-actions-hash.js

```diff
@@ -0,0 +1,60 @@
+const ember = require('../utils/ember');
+
+const ERROR_MESSAGE = 'Use the @action decorator instead of declaring an actions hash';
+
+module.exports = {
+  ERROR_MESSAGE,
+  meta: {
+    docs: {
+      description: 'Disallows the actions hash in components, controllers and routes',
+      category: 'Ember Object',
+      recommended: false,
+    },
+    fixable: null, // or "code" or "whitespace"
+    url:
+      'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-actions-hash.md',
+  },
+
+  create: context => {
+    const filePath = context.getFilename();
+    let inClassWhichCanContainActions = false;
+
+    function _inClassWhichCanContainActions(node, filePath) {
+      return (
+        inClassWhichCanContainActions ||
+        ember.isEmberComponent(node, filePath) ||
+        ember.isEmberController(node, filePath) ||
+        ember.isEmberRoute(node, filePath)
+      );
+    }
+
+    return {
+      ClassDeclaration(node) {
+        inClassWhichCanContainActions = _inClassWhichCanContainActions(node, filePath);
+      },
+      CallExpression(node) {
+        inClassWhichCanContainActions = _inClassWhichCanContainActions(node, filePath);
+      },
+      ObjectExpression(node) {
+        if (!inClassWhichCanContainActions) {
```

Comment:
I don't think this check would be safe/accurate if the file had multiple classes in it. Instead, this code should probably go in the `CallExpression` function.

```
CallExpression(node) {
  // if right kind of class, check to see if the node's body object contains `actions`
}
```

And same problem for `ClassDeclaration`/`ClassProperty` code.